src/main/java/org/junit/runners/BlockJUnit4ClassRunner.java

```diff
@@ -463,4 +441,22 @@ private long getTimeout(Test annotation) {
         }
         return annotation.timeout();
     }
+
+    private static final ThreadLocal<RuleContainer> CURRENT_RULE_CONTAINER =
+            new ThreadLocal<RuleContainer>();
```

Comment:
Why does this have to be a `ThreadLocal`? Can't we use an instance variable and pass it to the `RuleCollector` constructor?